pkg/cmd/kafka/create/create.go

```diff
@@ -507,11 +527,26 @@ func promptKafkaPayload(opts *options, userQuotaType accountmgmtutil.QuotaSpec)
 		Marketplace:           marketplaceProviderNullable,
 	}
 	printSizeWarningIfNeeded(opts.f, answers.Size, sizes)
-	payload.SetPlan(mapAmsTypeToBackendType(&userQuotaType) + "." + answers.Size)
+	payload.SetPlan(mapAmsTypeToBackendType(userQuotaType) + "." + answers.Size)
 
 	return payload, nil
 }
 
+func promptBillingModelSelect(localizer localize.Localizer, answers *promptAnswers) error {
+
+	billingModelPrompt := &survey.Select{
+		Message: "Billing model:",
+		Options: kafkacmdutil.ValidBillingModels,
```

Comment:
That will outdate fast as new billing models should be introduced. 
I think the best tradeoffs would be to return error to user from ams package containing billing models he can create when invalid is provided or billing model is missing.